iceberg/src/test/scala/org/apache/spark/sql/delta/ConvertToHudiSuite.scala

```diff
@@ -0,0 +1,224 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hudi.common.config.HoodieMetadataConfig
+import org.apache.hudi.common.engine.HoodieLocalEngineContext
+import org.apache.hudi.common.fs.FSUtils
+import org.apache.hudi.common.model.HoodieBaseFile
+import org.apache.hudi.common.table.{HoodieTableMetaClient, TableSchemaResolver}
+import org.apache.hudi.metadata.HoodieMetadataFileSystemView
+import org.apache.spark.SparkContext
+import org.apache.spark.sql.{QueryTest, SparkSession}
+import org.apache.spark.sql.avro.SchemaConverters
+import org.apache.spark.sql.types.StructType
+import org.apache.spark.util.Utils
+import org.scalatest.concurrent.Eventually
+import org.scalatest.time.SpanSugar._
+
+import java.io.File
+import java.time.Instant
+import java.util.stream.Collectors
+import scala.collection.JavaConverters
+
+class ConvertToHudiSuite extends QueryTest with Eventually {
+
+  private var _sparkSession: SparkSession = null
+  private val TMP_DIR = Utils.createTempDir().getCanonicalPath
+  private val testTableName: String = "deltatable"
+  private val testTablePath: String = s"$TMP_DIR/$testTableName"
+
+  override def spark: SparkSession = _sparkSession
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    _sparkSession = createSparkSession()
+  }
+
+  override def afterEach(): Unit = {
+    super.afterEach()
+    _sparkSession.sql(s"DROP TABLE IF EXISTS $testTableName")
+    Utils.deleteRecursively(new File(testTablePath))
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+    SparkContext.getActive.foreach(_.stop())
+  }
+
+  test("basic test - managed table created with SQL") {
+    _sparkSession.sql(
+      s"""CREATE TABLE `$testTableName` (col1 INT) USING DELTA
+         |LOCATION '$testTablePath'
+         |TBLPROPERTIES (
+         |  'delta.universalFormat.enabledFormats' = 'hudi'
+         |)""".stripMargin)
+    _sparkSession.sql(s"INSERT INTO `$testTableName` VALUES (123)")
+    verifyFilesAndSchemaMatch()
+  }
+
+  test("basic test - catalog table created with DataFrame") {
+    withDefaultTablePropsInSQLConf {
+      _sparkSession.range(10).write.format("delta")
+        .option("path", testTablePath)
+        .saveAsTable(testTableName)
+    }
+    _sparkSession.range(10, 20, 1)
+      .write.format("delta").mode("append")
+      .save(testTablePath)
+    verifyFilesAndSchemaMatch()
+  }
+
+  test("validate multiple commits (non-partitioned)") {
```

Comment:
plz refactor and combine with next test case. 

for (isPartitioned <- Seq(true, false)) {
   test("XXXX") {
     val partitionSQLStr = " PARTITIONED BY (col3) "
     s"""CREATE XXX $partitionSQLStr XXX """
  }
}

Reply:
Thanks! I was wondering how parameterized testing worked in scala. I have updated the test